validation/html_used.ts

```diff
@@ -14,30 +14,36 @@ export async function getHTMLUsed(
   client: CDPSession,
   page: Page
 ): Promise<{[implementation: string]: string}> {
-  const htmlUsedByChrome = await getHTMLUsedByChrome(nodeRef, client, page);
-  const htmlUsedByOurAccName = (await page.evaluate(
-    "ourLib.getAccessibleName(document.querySelector('" +
-      nodeRef.selector +
-      "')).visitedHTMLSnippet;"
-  )) as string;
-  return {chrome: htmlUsedByChrome, accname: htmlUsedByOurAccName};
+  const nodesUsedByChrome = await getNodesUsedByChrome(nodeRef, client, page);
+  const chromeHandles = nodesUsedByChrome.map(node => node.handle);
+  const htmlUsedByChrome = await getHTMLFromHandles(chromeHandles, page);
+
+  // Initialise nodeArray and get its length
+  const accnameNodeArrayLength = await page.evaluate(`
+    const nodeSet = accname.getNameComputationDetails(document.querySelector('${nodeRef.selector}')).nodesUsed;
+    const nodeArray = Array.from(nodeSet);
+    nodeArray.length;`);
+  // Make an array containing an ElementHandle for each Node in nodeArray
+  const accnameHandles = (await Promise.all(
+    [...Array(accnameNodeArrayLength).keys()].map(i =>
+      page.evaluateHandle(`nodeArray[${i}]`)
+    )
+  )) as ElementHandle<Element>[];
```

Comment:
🙁 getting the length then mapping an array of inegers through `page.evaluateHandle` 🙁 

What comes out of `page.evaluate` if you try to return `nodeArray`? I guess it doesn't deserialize `Element` to `ElementHandle`?

Reply:
We get a `JSHandle` containing an `Array<Element>` when we call `page.evaluateHandle()` on an array (an `Array<JSHandle>` would've been ideal here).

If we can get `getNodesUsedByChrome()` to return a `JSHandle<Array<Element>>` as well then we can modify the function `getHTMLFromHandles` to process both `chromeHandles` and `accnameHandles`.